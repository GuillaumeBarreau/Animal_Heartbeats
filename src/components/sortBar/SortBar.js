import React, { useState } from 'react';
import { connect } from 'react-redux';
import { FILTERBYNAME } from '../../store/actions/actionTypes';
// Import Styled components
import { ContentInput, Input } from './SortBar.styled';

const SortBar = ({ dispatch, theme }) => {
	const [inputValueState, setInputValueState] = useState('');

	const handleChange = event => {
		setInputValueState(event.target.value);
		dispatch({
			type: FILTERBYNAME,
			value: event.target.value,
		});
	};

	const props = {
		theme: theme,
		type: 'text',
		placeholder: 'Searching by name',
		onChange: handleChange,
		value: inputValueState,
	};

	return (
		<ContentInput>
			<Input {...props} />
		</ContentInput>
	);
};

const mapStateToProps = state => {
	return {
		theme: state.theme,
	};
};

export default connect(mapStateToProps)(SortBar);
