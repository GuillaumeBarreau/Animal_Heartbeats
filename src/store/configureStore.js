import { createStore, applyMiddleware } from 'redux';
import reducers from './reducers/combineReducers';

function logger({ getState }) {
	return next => action => {
		console.log('will dispatch', action);
		return next(action);
	};
}

export const store = createStore(reducers, applyMiddleware(logger));
 