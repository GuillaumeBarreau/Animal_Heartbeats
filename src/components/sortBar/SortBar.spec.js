import { shallow } from 'enzyme';
import React from 'react';
import SortBar from './SortBar';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

describe('SortBar component', () => {
	let store;

	beforeEach(() => {
		store = mockStore({
			theme: 'dark',
		});
	});

	it('Render SortBar component', () => {
		const cmp = shallow(
			<Provider store={store}>
				<SortBar />
			</Provider>
		);
		expect(cmp.exists()).toBe(true);
	});

	it('Snapshot testing', () => {
		const cmp = shallow(
			<Provider store={store}>
				<SortBar />
			</Provider>
		);
		expect(cmp.html()).toMatchSnapshot();
	});
});
