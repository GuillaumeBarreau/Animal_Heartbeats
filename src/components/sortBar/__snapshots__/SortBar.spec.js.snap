// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar component Snapshot testing 1`] = `"<div class=\\"sc-AykKC kkUZaI\\"><input type=\\"text\\" placeholder=\\"Searching by name\\" value=\\"\\" class=\\"sc-AykKD djKtFZ\\"/></div>"`;
